src/sentry/shared_integrations/client/base.py

```diff
@@ -329,3 +333,59 @@ def get_with_pagination(
             if num_results < page_size:
                 return output
         return output
+
+    def record_response(self, response: Response | None = None, error: Exception | None = None):
+        print("record_response")
+        if self.is_response_fatal(response, error):
+            self.record_request_fatal(response, error)
+        elif self.is_response_error(response, error):
+            self.record_request_error(response, error)
+        elif self.is_response_success(response):
+            self.record_request_success(response)
+
+    def record_request_error(self, resp: Response | None = None, error: Exception | None = None):
+        if not self._get_redis_key():
+            return
+        if not self.is_response_error(resp, error):
+            return
+        buffer = IntegrationRequestBuffer(self._get_redis_key())
+        buffer.record_error()
+        print("error recorded")
+        if buffer.is_integration_broken():
+            self.uninstall_integration(self.integration_id)
+
+    def record_request_success(self, resp: Response):
+        if not self._get_redis_key():
```

Comment:
I'd probably store the key in a variable so you don't have to call `_get_redis_key()` twice